my_account_helper/model/xmlrpc_helper.py

```diff
@@ -0,0 +1,47 @@
+﻿# -*- encoding: utf-8 -*-
+#
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emmanuel Mathier
+#
+#    The licence is in the file __openerp__.py
+#
+
+from openerp.osv import orm
+
+
+class xmlrpc_helper(orm.Model):
+
+    _inherit = "res.partner"
+
+    # return all child information for all childs from a partner_id specified
+    def get_childs_from_partner(self, cr, uid, partner_id, context=None):
+
+        # get the Singleton instance of the recurring.contract model from the
+        # registry pool for the database in use
+        contract_obj = self.pool.get('recurring.contract')
+
+        # get all contract_id from a partner_id
+        contract_ids = contract_obj.search(
+            cr, uid, [('partner_id', '=', partner_id)], context=None)
+
+        # get all child information from each contract_id for each child
+        childs = list()
+        for contract in contract_obj.browse(cr, uid, contract_ids, context):
+            child = dict()

```

Comment:
Not all contracts may be linked to a child, some contracts are not sponsorships. So you should first check if a child is linked to the contract,
```
if contract.child_id:
    child = dict()
    ...
```